src/app/component/explorer/index.vue

```diff
@@ -262,36 +301,70 @@ export default {
 
     updated () {
         this.resizeTable();
+        this.getDatabases();
     },
 
     methods: {
-        resizeTable () {
-            let thead, startOffset;
+        modalDump () {
+            this.openModal = true;
+        },
 
-            document.querySelectorAll("table th").forEach(th => {
-                th.style.position = 'relative';
+        closeModal () {
+            this.openModal = false;
+        },
 
-                var grip = document.createElement('div');
-                grip.classList.add('resizeTable');
-                grip.innerHTML = "&nbsp;";
+        getDatabases () {
+            this.knex.raw('show databases').then(database => {
+                this.databases = database[0].map(db => db.Database);
+            })
+        },
 
-                grip.addEventListener('mousedown', function (e) {
-                    thead = th;
-                    startOffset = th.offsetWidth - e.pageX;
-                });
+        executeQuery () {
+            let mysql = require('mysql');
+            var mysqlDump = require('mysqldump');
+            let fs = require('fs');
+            let os = require('os');
+            let file = document.getElementById('sql').files[0];
+
+            let connection = mysql.createConnection({
+                host     : 'localhost',
+                user     : 'root',
+                password : '',
+                database : this.databaseActive,
+                multipleStatements: true,
+            });
 
-                th.appendChild(grip);
+            if (this.selectedDatabase) {
+                let homeDir = os.homedir();
+
+                mysqlDump({
+                    host: 'localhost',
+                    user: 'root',
+                    password: '',
+                    database: this.selectedDatabase,
+                    dest: `${homeDir}/Downloads/${this.selectedDatabase.toLowerCase()}.sql`
```

Comment:
What do you think about choose your destination path?

Hardcoded path it is not the better solution.